child_compassion/model/child_compassion.py

```diff
@@ -436,53 +513,186 @@ def child_sponsored(self, cr, uid, ids, context=None):
+        child.write({
+            'exit_date': json_data['exitDate'],
+            'last_attended_project': json_data['dateLastAttendedProject'],
+            'presented_gospel': json_data['presentedWithGospel'],
+            'professes_faith': json_data['professesFaithInJesusChrist'],
+            'faith_description': json_data['faithDescription'],
+            'primary_school': json_data['completedPrimarySchool'],
+            'us_grade_completed': json_data['usGradeEquivalentCompleted'],
+            'study_area': json_data['areaOfStudy'],
+            'vocational_training': json_data['receivedVocationalTraining'],
+            'vocational_skills': json_data['vocationalSkillsLearned'],
+            'disease_free': json_data['freeOfPovertyRelatedDisease'],
+            'health_description': json_data['healthDescription'],
+            'social_description': json_data['socialBehaviorDescription'],
+            'exit_description': json_data['exitDescription'],
+            'steps_prevent_description': json_data['stepsToPrevent'
+                                                   'ExitDescription'],
+            'future_plans_description': json_data['futurePlansDescription'],
+            'new_situation_description': json_data['childNewSituation'
+                                                   'Description'],
+            'exit_reason': json_data['exitReason'],
+            'last_letter_sent': json_data['lastChildLetterSent'],
+        })
 
-    def child_remove_from_typo3(self, cr, uid, ids, context=None):
-        child_codes = [child.code for child in self.browse(cr, uid, ids,
-                                                           context)]
-        filename = "upd.sql"
-        file_query = open(filename, "wb")
-        for code in child_codes:
-            file_query.write(
-                "delete from tx_drechildpoolmanagement_domain_model_children "
-                "where child_key='%s';\n" % code)
-        file_query.close()
-        host = config.get('typo3_host')
-        username = config.get('typo3_user')
-        pwd = config.get('typo3_pwd')
-        scripts_url = config.get('typo3_scripts_url')
-        path = config.get('typo3_scripts_path')
-        api_key = config.get('typo3_api_key')
-        if not (host and username and pwd and path and scripts_url
-                and api_key):
-            raise orm.except_orm('ConfigError',
-                                 'Missing typo3 settings '
-                                 'in conf file')
-        with pysftp.Connection(host, username=username, password=pwd) as sftp:
-            with sftp.cd(path):
-                sftp.put(filename)
+        return True
+
+    def _get_typo3_child_id(self, cr, uid, child_code):
+        res = Sync_typo3.request_to_typo3(
+            "select * "
+            "from tx_drechildpoolmanagement_domain_model_children "
+            "where child_key='%s';" % child_code, 'sel')
+
+        return json.loads(res)[0]['uid']

```

Comment:
Are you sure that there will always be at least 1 element containing the hid key? Otherwise, surround with try except